inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -17,55 +17,241 @@
+		super.renderHead(aResponse);
+
+		// import Js
+		aResponse.render(JavaScriptHeaderItem.forReference(new WebjarsJavaScriptResourceReference("c3/current/c3.js")));
+		aResponse.render(
+				JavaScriptHeaderItem.forReference(new WebjarsJavaScriptResourceReference("d3js/current/d3.js")));
+
+		// import Css
+		aResponse.render(CssHeaderItem.forReference(new WebjarsCssResourceReference("c3/current/c3.css")));
+	}
+
+	@OnEvent
+	public void onRenderAnnotations(RenderAnnotationsEvent aEvent) {
+		try {
+			log.info("rendered annotation event");
+
+			StringBuilder chartType = new StringBuilder();
+			
+			HashMap<String, List<Double>> recommenderScoreMap = getLatestScores(aEvent);
+
+			// iterate over the recommender score map to create data arrays to feed to the
+			// c3 graph
+			StringBuilder dataColumns = new StringBuilder();
+			Iterator<Entry<String, List<Double>>> it = recommenderScoreMap.entrySet().iterator();
+			while (it.hasNext()) {
+				Entry<String, List<Double>> pair = it.next();
+
+				String jsonString = toJsonString(pair);
+				jsonString = jsonString.replaceAll("\".*?\"", "");
+				jsonString = jsonString.replaceAll("'.*?'", "");
+				jsonString = jsonString.replaceAll("`.*?`", "");
+
+				String data = substring(jsonString, 3, -2);
+
+				// append recommender name
+				dataColumns.append("['");
+				String[] recommenderClass = pair.getKey().toString().split("\\.");
+				String recommenderName = recommenderClass[recommenderClass.length - 1];
+				
+				//define chart type for the recommender
+				chartType.append(recommenderName);
+				chartType.append(": 'step', ");
+				
+				dataColumns.append(recommenderName);
+
+				// append data columns
+				dataColumns.append("', ");
+				dataColumns.append(data);
+				dataColumns.append("]");
+				dataColumns.append(",");
+
+				// avoids a ConcurrentModificationException
+				it.remove();
+			}
+
+			String javascript = createJSScript(dataColumns.toString(), chartType.toString());
+			log.info(javascript);
+
+			aEvent.getRequestHandler().prependJavaScript(javascript);
+		} catch (IOException e) {
+			log.error("Unable to render chart", e);
+			error("Unable to render chart: " + e.getMessage());
+			aEvent.getRequestHandler().addChildren(getPage(), IFeedback.class);
+		}
+	}
+
+	/**
+	 * Creates the js script to render graph with the help of given data points. Sample value of dataColumns:
+	 * ['recommender1', 1.0, 2.0, 3.0 ], ['recommender2', 2.0, 3.0, 4.0]. Also creates an xaxix of a sequence
+	 * from 0 to maximumNumberOfPoints (50)
+	 * 
+	 * @param dataColumns
+	 * @param chartType 
+	 * @return
+	 * @throws IOException
+	 */
+	private String createJSScript(String dataColumns, String chartType) throws IOException {
+		int[] intArray = IntStream.range(0, maxPointsToPlot).map(i -> i).toArray();
+
+		String xaxisValues = "[ 'x' ," + substring(Arrays.toString(intArray), 1, -1) + "]";
+		String data = toJsonString(dataColumns).substring(1, dataColumns.toString().length());
+
```

Comment:
Newline too much

Reply:
You have two newlines here, you need at max one.

Reply:
Got it.